FRPs/proposal-2-liquidations.md

```diff
@@ -0,0 +1,50 @@
+
+One dominant liquidation design is the fixed spread liquidation mechanism, used by Aave, Compound and dYdX. Interestingly, the fraction of the collateral that a liquidator can purchase at a discount amounts to 50% for Aave and Compound, and even 100% for dYdX (commonly referred to as close factor). It therefore appears that the fixed spread liquidation mechanisms are benefitting the liquidators more than the borrowers.
+
+Drawing the link to MEV, each liquidation event may qualify as Miner Extractable Value (MEV). For example, in a recent price oracle manipulation attack, a liquidator performed an oracle price post and liquidated millions of USD within one transaction, yielding a profit of nearly 4M USD. 
+
+The goal of this FRP is to offer a systematization of existing liquidation mechanisms and consistent measurements of the past and current DeFi liquidation state. We aim to investigate alternative liquidation designs in an effort to mitigate MEV from liquidation events.
+
+## Background and Problem Statement
+
+The goal of this FRP is to study the past and current status of blockchain liquidation. Specifically, we would like to answer the following questions:
+- What is the monthly accumulated liquidation profit for Aave, Compound, dYdX, MakerDAO?
+- What gas price distributions are being paid for the liquidation events?
+- What are the different liquidation methods, and how do they differ with respect to MEV? Which liquidation mechanism avoids MEV?
+- How sensitive are current debts to liquidation (by how much would an adversary need to manipulate a price oracle)? An MEV miner could try to manipulate an oracle to extract this MEV.
+- How can we design a possible liquidation mechanism that mitigates MEV?
+
+## Plan and Deliverables
+
+- Our report is envisioned to contain multiple figures/quantitative results regarding: 
+  - Monthly liquidation profits for MakerDAO/Aave/Compound/dYdX.
+  - Understand the gas prices of liquidation events. Is there a way to avoid gas fee bidding contests in liquidations in an effort to preserve valuable block-space?
+  - Quantify the liquidation sensitivity (how much collateral is liquidated given a % price decline of the collateral) for MakerDAO/Aave/Compound/dYdX and most of the assets on these platforms.
+  - How many liquidation events are profitable to the liquidator?
+  - How many liquidations make use of flash loans?
+ 
+- We plan to systematize the different liquidation mechanisms in an effort to understand their strengths and weaknesses.
+- We would love to integrate with the inspectors of the flashbots dashboard to crawl our data. Yet, we haven’t been able to look into further technical details of how they work and what the current architecture is. If too complicated we can also proceed with our current geth-based infrastructure.
```

Comment:
Could you share the geth infra by any chance? Geth's tracing is very very low performance, so any trace analysis you'd do via that would be super slow. Traces are a must if you want to get into liquidations that do fancier stuff than just vanilla `liquidateBorrow` calls. I broke down the [mev-inspect ](https://github.com/flashbots/mev-inspect-rs/) architecture [here](https://discord.com/channels/755466764501909692/755466928637870222/787923691265589270). It is quite modular and high perf (analyzes a full block in ~0.2s)

Reply:
We can eventually merge open archival sources (I'm thinking BigQuery to start since pulling traces out of there is cheap) and I agree the deliverable here should be something maintained by the mev-inspect team.

Reply:
For liquidations it appears that we don't have to use `trace` but can simply rely on `events`, or would you disagree? There's then no particular infrastructure necessary besides using geth's API to filter for certain events.